gerritlab/main.py

```diff
@@ -126,92 +121,78 @@ def cancel_prev_pipelines(commit):
                 local_branch, utils.get_change_id(commits[idx - 1].message))
         commits_data.append(Commit(c, source_branch, target_branch))
 
-    # Before we update the existing the MRs or create new MRs, there are a few
-    # important notes:
-    #
-    # 1) If an MR's source branch becomes a subset of its target branch, that
-    #    is, commits in the source branch are all included in the target branch,
-    #    the MR will be auto-merged as it becomes meaningless.
-    # 2) Because of 1, we must be careful when updating the existing MRs. For
-    #    example, there are 3 existing MRs mr0, mr1 and mr2, with their
-    #    dependencies as master <= mr0 <= mr1 <= mr2. If we locally rebase the
-    #    commits to be master <= mr2 <= mr0 <= mr1, pushing the new commits to
-    #    each MR's source branch will end up auto-merging mr2, as mr'2 target
-    #    branch mr1 now contains its commit. To avoid this, we must update an
-    #    MR's target branch before pushing the new commit. Also, we must update
-    #    MRs from the end of the existing MR dependency chain. In this example,
-    #    we must update mr2's target branch first and then push the new commit.
-    #    The same goes for mr0 and mr1.
-    #
-    # The following code updates/creates MRs with the above notes in mind.
+    push_time = 0
+    get_mrs_time = 0
+    update_mr_time = 0
+    create_mr_time = 0
 
+    # New approach
+    # Single push w/ all branch updates.
+    # Then create or update MRs.
+
+    # Push commits to Change-Id-named branches
+    refs_to_push = ["{}:refs/heads/{}".format(
+        c.commit.hexsha, c.source_branch) for c in commits_data]
+    print(f"Pushing {refs_to_push}")
+    start = time.time()
+    remote.push(refspec=refs_to_push, force=True)        
```

Comment:
This is great! I didn't even know you can push to multiple branches in a single command.

Reply:
That was a recent discovery myself.